src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,624 @@
+ * <h3>Implementation Requirements</h3>
+ * This class is immutable and thread-safe.
+ * <p>
+ * This class must be treated as a value type. Do not synchronize, rely on the
+ * identity hash code or use the distinction between equals() and ==.</p>
+ */
+public final class InternationalFixedDate
+        extends AbstractDate
+        implements ChronoLocalDate, Serializable {
+
+    /**
+     * Serialization version UID.
+     */
+    private static final long serialVersionUID = -7473722012599657263L;
+
+    /**
+     * The days per 400 year cycle.
+     */
+    private static final int DAYS_PER_CYCLE = 146097;
+
+    /**
+     * Number of years in a decade.
+     */
+    private static final int YEARS_IN_DECADE = 10;
+
+    /**
+     * Number of years in a century.
+     */
+    private static final int YEARS_IN_CENTURY = 100;
+
+    /**
+     * Number of years in a millennium.
+     */
+    private static final int YEARS_IN_MILLENNIUM = 1000;
+
+    /**
+     * The proleptic year.
+     */
+    private final int prolepticYear;
+
+    /**
+     * The month.
+     */
+    private final short month;
+
+    /**
+     * The day.
+     */
+    private final short day;
+
+    /**
+     * In a leap year, is it the day between June 28th and Sol 1st ?
+     */
+    private final boolean isLeapDay;
+
+    /**
+     * Is is the last day of the year ?
+     */
+    private final boolean isYearDay;
+
+    //-----------------------------------------------------------------------
+
+    /**
+     * Creates an instance from validated data.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from 1 to 13
+     * @param dayOfMonth    the International fixed day-of-month, from 1 to 28, the 29th is only legal for leap day and year day
+     */
+    private InternationalFixedDate (final int prolepticYear, final int month, final int dayOfMonth) {
+        ChronoField.YEAR.checkValidValue (prolepticYear);
+        InternationalFixedChronology.MONTH_OF_YEAR_RANGE.checkValidValue (month, ChronoField.MONTH_OF_YEAR);
+        InternationalFixedChronology.DAY_OF_MONTH_RANGE.checkValidValue (dayOfMonth, ChronoField.DAY_OF_MONTH);
+
+        if (1 > prolepticYear) {
+            throw new DateTimeException ("Invalid date, year must be at least 1: " + prolepticYear + '-' + month + '-' + dayOfMonth);
+        }
+
+        if (dayOfMonth > 28 && month != 6 && month != 13) {
+            throw new DateTimeException ("Invalid date: " + prolepticYear + '-' + month + '-' + dayOfMonth);
+        }
+
+        this.prolepticYear = prolepticYear;
+        this.month = (short) month;
+        this.day = (short) dayOfMonth;
+        this.isLeapDay = dayOfMonth == 29 && month == 6;
+        this.isYearDay = dayOfMonth == 29 && month == 13;
+
+        if (this.isLeapDay && !isLeapYear ()) {
+            throw new DateTimeException ("Invalid leap date: " + prolepticYear + '-' + month + '-' + dayOfMonth);
```

Comment:
This message should probably state the given year isn't actually a leap year.  It would be better to pass `prolepticYear` to the static method in the chronology, too.